src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1754,6 +1754,11 @@ protected internal virtual TypeDefinition MarkType (TypeReference reference, Dep
 			foreach (Action<TypeDefinition> handleMarkType in _markContext.MarkTypeActions)
 				handleMarkType (type);
 
+			if (type.BaseType != null &&
+				!_context.Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (type) &&
+				_context.Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (_context.TryResolve (type.BaseType)))
+				CheckAndReportRequiresUnreferencedCode (type);
```

Comment:
With the discussion from https://github.com/mono/linker/issues/2136 in mind - I think we should use a different warning code in this case. The reason is that otherwise it's tricky to suppress this warning correctly.

The warning should be reported on the derived type itself, so suppressing it there is the only viable option, but that also means that the suppression is effective for everything in the type. If this were just `IL2026` suppressing it would suppress all `IL2026` warnings in the type and in all its methods. We should use a new warning code just for this case so that suppressing it is more precise. It is still not 100% precise as it would suppress the warning on all nested types as well, but that's probably good enough.